MeshLib/MeshEditing/ElementValueModification.h

```diff
@@ -37,13 +41,40 @@ class ElementValueModification
 	/// Returns true if successful or false if the value is already taken.
 	static bool replace(MeshLib::Mesh &mesh, unsigned old_value, unsigned new_value, bool replace_if_exists = false);
 
+	static bool replace(MeshLib::Mesh &mesh, std::string const& property_name,
+		unsigned old_value, unsigned new_value, bool replace_if_exists = false);
+
 	/// Sets new value for all elements having the given element type
 	/// Returns the number of elements having the given element type
 	static unsigned setByElementType(MeshLib::Mesh &mesh, MeshElemType ele_type, unsigned new_value);
 
 private:
-	/// Returns the values of elements within the mesh
-	static std::vector<unsigned> getMeshValues(const MeshLib::Mesh &mesh);
+	/// Returns sorted values of properties within the PropertyVector
+	/// These values are stored in a vector.
+	template <typename T>
+	static std::vector<T> getSortedPropertyValues(
+		MeshLib::PropertyVector<T> const& property_vector)
+	{
+		std::vector<T> value_mapping;
+		const std::size_t n_property_values(property_vector.size());
+		for (std::size_t i=0; i<n_property_values; ++i) {
+			bool exists(false);
+			T const& value (property_vector[i]);
+			std::size_t const size(value_mapping.size());
+			for (unsigned j=0; j<size; ++j) {
```

Comment:
While the number of property values is usually quite big, I expect a small number of different values here. The above implementation takes O(n) x O(m), where n is the total number of properties and m is the number of different property values.

If my understanding of the suggestion is correct 
(1) a copy of the values values is needed (O(n))
(2) sorting (O(n log n))
(3) erasing duplicate values (O(n))

If the number of different property values is in the order of O(log n) the implementations should have almost the same run time. So all depends on the concrete case.